File: Algorithms/src/main/java/com/technicalyorker/algorithms/sorting/BubbleSort.java
package com.technicalyorker.algorithms.sorting;

import com.technicalyorker.algorithms.utilities.Util;

public class BubbleSort implements Sortable {

	public void sort(int[] a) {
		int N = a.length;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N - 1; j++) {
				if (a[j] > a[j + 1]) {
					Util.swap(a, j + 1, j);
				}
			}
		}
	}
}

File: Scala/scripts/bubble_sort.scala
import scala.collection.mutable.MutableList;
case class BubbleSort(val as:List[Int]) {
	def sort():MutableList[Int]= {
		var acopy =MutableList[Int]() 
		for(a <- as) acopy:+=a
		for(i<-(acopy.length-1) until 0 by -1) {
			for(j<-0 until i) {
				if(acopy(j)>acopy(j+1)) {
					Util.swap(acopy,j,j+1);
				}
			}
		}
		acopy
	}
	def main(args: Array[String]): Unit = {
	  
	}
}
object Util {
	def swap(l:MutableList[Int],x:Int,y:Int):Unit={
		var temp = l(x);
		l(x) = l(y);
		l(y) = temp;
	}
}
var b = new BubbleSort(List(3,4,6,2,1,9,6,0))
print(b.sort)
